string_manipulation.rb

```diff
@@ -1,30 +1,133 @@
+# Helpers
+def swap_in_place(string, left, right) #time complexity = O(c), space = O(c) (temp_var)
+  temp_var = string[left]
+  string[left] = string[right]
+  string[right] = temp_var
+end
+
+
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  puts "NOT IMPLEMENTED"
+  return my_string if my_string.length < 2
+
+  i = 0
+
+  while i < my_string.length/2
+    swap_in_place(my_string, i, my_string.length-i-1)
+    i +=1
+  end
+
+  return my_string
 end
 
 # A method to reverse each word in a sentence, in place.
-def reverse_words(my_words)
-  puts "NOT IMPLEMENTED"
+def reverse_words(my_words) #with string_reverse
+  return my_words if my_words.length < 2
+  return string_reverse(my_words) if !my_words.include?(" ")
+
+  first = 0
+
+  my_words.length.times do |last |
+    if my_words[last].match(/\s/)
+      my_words[first...last] = string_reverse(my_words[first...last])

```

Comment:
Since everything is happening in place, you don't need to assign. i.e. the following would do the trick:
` swap_in_place(my_words, first, last)`